src/util/retry.rs

```diff
@@ -80,6 +80,36 @@ impl<'a, R: Read> Read for RetryReader<'a, R> {
     }
 }
 
+/// A Read that will only read up to a given number of bytes before EOF'ing.
+pub struct ReadCounter<'a, R: Read> {
```

Comment:
How is this different from `BoundReader`?

Reply:
This is unbounded (except by u64) for just the purpose of determining how many bytes were read. The savings over using BoundReader are probably minimal, and so I could just use BoundReader with u64::max() instead.

Reply:
It seems that `ReadCounter` is only used once (in the mempool acceptance logic), and it looks like we should be using a `BoundReader` there instead in order to avoid an "infinitely big" transaction DDoS.  Maybe this struct doesn't need to exist after all?

Reply:
Yeah, I'll remove and replace with a bound reader. I assumed that the network code would wrap the reader in a bound reader, and didn't want to end up wrapping the same stream multiple times. I'll wrap the stream in a bound reader and add a note that the input stream does not need to be bounded.

Reply:
The network code will wrap the inbound data stream with a BoundReader for the maximum message length, but I think for transactions, we want this bound to be smaller (i.e. the maximum size of a transaction is strictly less than the maximum size of a block or microblock, since the maximum block size includes its header, and the maximum message size includes the data and the message preamble).

Reply:
Addressed in ca8eacd -- I added a bound reader to the deserialization method of StacksTransaction